rollup/internal/controller/watcher/chunk_proposer.go

```diff
@@ -194,10 +188,6 @@ func (p *ChunkProposer) proposeChunk() error {
 	}
 
 	maxBlocksThisChunk := p.maxBlockNumPerChunk
-	blocksUntilFork := forks.BlocksUntilFork(unchunkedBlockHeight, p.forkHeights)
-	if blocksUntilFork != 0 && blocksUntilFork < maxBlocksThisChunk {
-		maxBlocksThisChunk = blocksUntilFork
-	}
```

Comment:
if we remove this, a chunk may include blocks from different forks.

Reply:
it's replaced by the following logic:
```
	// Ensure all blocks in the same chunk use the same codec version
	// If a different codec version is found, truncate the blocks slice at that point
	for i := 1; i < len(blocks); i++ {
		currentCodecVersion := getCodecVersion(p.chainCfg, blocks[i].Header.Number.Uint64(), blocks[i].Header.Time)
		if currentCodecVersion != codecVersion {
			blocks = blocks[:i]
			maxBlocksThisChunk = uint64(i) // update maxBlocksThisChunk to trigger chunking, because these blocks are the last blocks before the hardfork
			break
		}
	}
```

Reply:
fixed in this commit: [5cde900](https://github.com/scroll-tech/scroll/pull/1397/commits/5cde9004168ae9ae2fee0c59f00418ae5e6ef35d)